lambdas/bucket-setup/index.py

```diff
@@ -0,0 +1,58 @@
+import boto3
+import cfnresponse
+
+s3_client = boto3.client('s3')
+
+def handler(event, context):
+    if event['RequestType'] == 'Delete':
+        cfnresponse.send(event, context, cfnresponse.SUCCESS, {})
+        return
+
+    try:
+        props = event['ResourceProperties']
+        bucket = props['Bucket']
+        catalog_host = props['QuiltWebHost']
+        enable_versioning(bucket)
+        set_cors(bucket, catalog_host)
+        cfnresponse.send(event, context, cfnresponse.SUCCESS, {})
+        return
+    except Exception as e:
+        cfnresponse.send(event, context, cfnresponse.FAILED, {})
+        raise
+
+def enable_versioning(bucket):
+    s3_client.put_bucket_versioning(
+            Bucket=bucket,
+            VersioningConfiguration={
+                'Status': 'Enabled'
+            }
+        )
+
+def set_cors(bucket, catalog_host):
+    new_cors_rule = {
+        'AllowedHeaders': [ '*' ],
+        'AllowedMethods': [
+            'GET',
+            'HEAD',
+            'PUT',
+            'POST'
```

Comment:
Intentionally leaving out DELETE?

Reply:
the CORS policy Aneesh specified left out DELETE. I assume we don't support any DELETEs from the catalog at the moment, and it's better to be safe I guess